Mathlib/Order/Disjoint.lean

```diff
@@ -0,0 +1,635 @@
+
+variable (c)
+
+theorem Codisjoint.sup_left (h : Codisjoint a b) : Codisjoint (a ⊔ c) b :=
+  h.mono_left le_sup_left
+#align codisjoint.sup_left Codisjoint.sup_left
+
+theorem Codisjoint.sup_left' (h : Codisjoint a b) : Codisjoint (c ⊔ a) b :=
+  h.mono_left le_sup_right
+#align codisjoint.sup_left' Codisjoint.sup_left'
+
+theorem Codisjoint.sup_right (h : Codisjoint a b) : Codisjoint a (b ⊔ c) :=
+  h.mono_right le_sup_left
+#align codisjoint.sup_right Codisjoint.sup_right
+
+theorem Codisjoint.sup_right' (h : Codisjoint a b) : Codisjoint a (c ⊔ b) :=
+  h.mono_right le_sup_right
+#align codisjoint.sup_right' Codisjoint.sup_right'
+
+variable {c}
+
+theorem Codisjoint.of_codisjoint_sup_of_le (h : Codisjoint (a ⊔ b) c) (hle : c ≤ a) :
+    Codisjoint a b :=
+  @Disjoint.of_disjoint_inf_of_le αᵒᵈ _ _ _ _ _ h hle
+#align codisjoint.of_codisjoint_sup_of_le Codisjoint.of_codisjoint_sup_of_le
+
+theorem Codisjoint.of_codisjoint_sup_of_le' (h : Codisjoint (a ⊔ b) c) (hle : c ≤ b) :
+    Codisjoint a b :=
+  @Disjoint.of_disjoint_inf_of_le' αᵒᵈ _ _ _ _ _ h hle
+#align codisjoint.of_codisjoint_sup_of_le' Codisjoint.of_codisjoint_sup_of_le'
+
+end SemilatticeSupTop
+
+section DistribLatticeTop
+
+variable [DistribLattice α] [OrderTop α] {a b c : α}
+
+@[simp]
+theorem codisjoint_inf_left : Codisjoint (a ⊓ b) c ↔ Codisjoint a c ∧ Codisjoint b c := by
+  simp only [codisjoint_iff, sup_inf_right, inf_eq_top_iff]
+  rfl
+
+@[simp]
+theorem codisjoint_inf_right : Codisjoint a (b ⊓ c) ↔ Codisjoint a b ∧ Codisjoint a c := by
+  simp only [codisjoint_iff, sup_inf_left, inf_eq_top_iff]
+  rfl
+
+theorem Codisjoint.inf_left (ha : Codisjoint a c) (hb : Codisjoint b c) : Codisjoint (a ⊓ b) c :=
+  codisjoint_inf_left.2 ⟨ha, hb⟩
+#align codisjoint.inf_left Codisjoint.inf_left
+
+theorem Codisjoint.inf_right (hb : Codisjoint a b) (hc : Codisjoint a c) : Codisjoint a (b ⊓ c) :=
+  codisjoint_inf_right.2 ⟨hb, hc⟩
+#align codisjoint.inf_right Codisjoint.inf_right
+
+-- porting note: the mathlib3 proof did not need `a` to be provided explicitly
+theorem Codisjoint.left_le_of_le_inf_right (h : a ⊓ b ≤ c) (hd : Codisjoint b c) : a ≤ c :=
+  @Disjoint.left_le_of_le_sup_right αᵒᵈ _ _ _ a _ h hd.symm
+#align codisjoint.left_le_of_le_inf_right Codisjoint.left_le_of_le_inf_right
+
+theorem Codisjoint.left_le_of_le_inf_left (h : b ⊓ a ≤ c) (hd : Codisjoint b c) : a ≤ c :=
+  hd.left_le_of_le_inf_right <| by rwa [inf_comm]
+#align codisjoint.left_le_of_le_inf_left Codisjoint.left_le_of_le_inf_left
+
+end DistribLatticeTop
+
+end Codisjoint
+
+open OrderDual
+
+theorem Disjoint.dual [SemilatticeInf α] [OrderBot α] {a b : α} :
+    Disjoint a b → Codisjoint (toDual a) (toDual b) :=
+  id
+#align disjoint.dual Disjoint.dual
+
+theorem Codisjoint.dual [SemilatticeSup α] [OrderTop α] {a b : α} :
+    Codisjoint a b → Disjoint (toDual a) (toDual b) :=
+  id
+#align codisjoint.dual Codisjoint.dual
+
+@[simp]
+theorem disjoint_to_dual_iff [SemilatticeSup α] [OrderTop α] {a b : α} :
```

Comment:
```suggestion
theorem disjoint_toDual_iff [SemilatticeSup α] [OrderTop α] {a b : α} :
```
Also below. These will need #align. (I'd recommend leaving in all #aligns, even when mathport gets the names right; easier when you have to put one back in later!)